### ISSUE TITLE
unicode-strings by saving dicts into hdfarchives

### ISSUE THREAD

**Issue by MHarland** [TRIQS/triqs#271]
If you save a BlockGf as part of a dictionary into HDFArchives, then loading it again you create a unicode string, that e.g. conflicts with the %s-operator in block_gf.py line 92. So in the example below only the third case produces that error. A quick workaround could be an additional string conversion changing line 92 to
```
for i,g in self: g.name = "%s_%s"%(str(self.name),i) if self.name else '%s'%(i,)
```
but you probably want to avoid unicode strings at all and the proper fix has to be applied somewhere else.

Error-code:
```
from pytriqs.gf.local import BlockGf, GfImFreq
from pytriqs.archive import HDFArchive

g = BlockGf(name_list = ['up','dn'], 
            block_list = [GfImFreq(indices = [0], beta = 10)]*2,
            name = 'testG',
            make_copies = True)

with HDFArchive('test.h5', 'w') as arch:
    arch.create_group('gr')
    arch['g1'] = g
    arch['gr']['g2'] = g
    arch['d'] = {'g3':g}

with HDFArchive('test.h5', 'r') as arch:
    g1 = arch['g1']
    g2 = arch['gr']['g2']
    g3 = arch['d']['g3']
```

A related issue could be #170 .

**Comment by krivenko** [comment]
The line `g3 = arch['d']['g3']` leads to a call `BlockGf.__factory_from_dict__(cls, name, d)` with `name` being a Unicode string (`u'g3'` in this example).

However, it is unclear to me what function should be responsible for converting `name` to a normal string. Must `BlockGf.__factory_from_dict__()` take care of it?
Otherwise it must be done inside `hdf_archive.py`, meaning that `__factory_from_dict__(key,values)` is never called with Unicode keys (for any class).

**Comment by parcollet** [comment]
Same fix was applied in ad2efbfaf8ab1c3f3103b6eade05a6a4d8b33b9b
Closing.
